resolvers.py

```diff
@@ -0,0 +1,1979 @@
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        # filters variable takes capability fields than needs to be shown
+        filters = kwargs.get("filters")
+        qs = self.add_filter(filters)
+        for element in qs:
+            element["band_name"] = "Total"
+            element["capability_name"] = "Total"
+            element["count_percentage"] = round(element["count_percentage"], 2)
+        return qs
+
+
+class LastUpdateTimeResolver:
+    def __call__(self, source, info, **kwargs):
+        return models.LastUpdatedTime.objects.first()
+
+
+class AllocatedHasherHelper:
+    '''Helper for allocated hasher (cell download) '''
+
+    def duration_filter_helper(self, duration, first_qs_row):
+
+        if first_qs_row.week_1 == duration:
+            qs_duration_filter = Q(employment_type_for_week_1='FTE')
+
+        elif first_qs_row.week_2 == duration:
+            qs_duration_filter = Q(employment_type_for_week_2='FTE')
+
+        elif first_qs_row.week_3 == duration:
+            qs_duration_filter = Q(employment_type_for_week_3='FTE')
+
+        elif first_qs_row.week_4 == duration:
+            qs_duration_filter = Q(employment_type_for_week_4='FTE')
+
+        elif first_qs_row.week_5 == duration:
+            qs_duration_filter = Q(employment_type_for_week_5='FTE')
+
+        elif first_qs_row.week_6 == duration:
+            qs_duration_filter = Q(employment_type_for_week_6='FTE')
+
+        elif first_qs_row.week_7 == duration:
+            qs_duration_filter = Q(employment_type_for_week_7='FTE')
+
+        elif first_qs_row.week_8 == duration:
+            qs_duration_filter = Q(employment_type_for_week_8='FTE')
+
+        else:
+            return False
+
+        return qs_duration_filter
+
+    def error_helper(self, error_message):
+        return {"ok": False, "error": f"The error is -> {error_message}", "data": []}
+
+    def cell_download_helper(self, pod_id, project_name, account_name, billability, duration, start_date=None, end_date=None, report_name=None):
+
+        try:
+            # calculating required dates
+            now = date.today()
+            current_monday = now - timedelta(days=now.weekday())
+            current_week_sunday = now + timedelta(days=(6 - now.weekday()))
+
+            if pod_id in ['Investment', 'Oversight Allocated']:
+                qs_pod_id_filter = Q(derived_allocation_type=pod_id)
+            else:
+                qs_pod_id_filter = Q(pod_id=pod_id)
+
+            # creating filter of project ,account name , billability and WBS code
+            qs_project_filter = Q(project_name=project_name,
+                                  account_name=account_name,
+                                  derived_billability=billability) & ~Q(project_wbs_code__startswith='CED')
+
+            # check report_name is present and return allocation filter accordingly
+            if report_name == None:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `==` and `!=` operators use the compared objects&#39; ` __eq__` method to test if they are equal. To check if an object is a singleton, such as `None`, we recommend that you use the `is` identity comparison operator.

[Learn more](https://docs.python.org/3/reference/expressions.html#is)

Similar issue at line number 1031.